lib/ukargparse/argparse.c

```diff
@@ -69,6 +69,12 @@ int uk_argnparse(char *argb, __sz maxlen, char *argv[], int maxcount)
 			if (!in_quote) {
 				argb[i] = '\0';
 				prev_wspace = 1;
+				break;
+			}
+			if (prev_wspace) {
```

Comment:
This is exactly what is missing, you are right. I wonder if we shouldn't add another label to the default case (e.g., `regularchar:`):
```diff
		default:
+		regularchar:
-			/* any character */
+			/* regular character */
			if (prev_wspace) {
				argv[argc++] = &argb[i];
				prev_wspace = 0;
			}
			break;
```
...and to use `goto` instead of the copied if-block and last `break`:
```c
		/* white spaces */
		case ' ':
		case '\r':
		case '\n':
		case '\t':
		case '\v':
			if (!in_quote) {
				argb[i] = '\0';
				prev_wspace = 1;
				break;
			}
			goto regularchar;
```
I think this would make the code much more readable and understandable. We also would avoid duplicated code in the function. Obviously I would also replace the `/* Fall through */` in the quotes case with the same `goto regularchar` jump.
What do you think?

Reply:
I think you're right, it looks a lot better. I will update the commit.